### PR TITLE
Adding new git action for updating OpenMS package numbers

### PR DESCRIPTION
**Pull request by JeeH-K**
## Description

Adding new git action. (Issue #6710)
1. Command "sed" seems to be os specific in some cases, and I've only tested with MacOS. Hence "runs-on" is on macos.
2. git configuration for commit should be updated

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: .github/workflows/update_version_numbers.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      # update INIs in tests topp:
+      find ./src/tests/topp/ -type f -name '*.ini' -exec grep "<ITEM name=\"version\" value=\".*\" type=\"string\"" {} \; -exec sed -i '' "s#<ITEM name=\"version\" value=\".*\" type=\"string\"#<ITEM name=\"version\" value=\"$package_version\" type=\"string\"#g" {} \;
+
+
+  commit_updated_files: # if any file is not updated, update files and push a commit
+    needs: updating_versions
+
+    steps:
+# Get updated files
+      - name: Get changed files
```
Comment (jpfeuffer) [inline .github/workflows/update_version_numbers.yml:38]: I think simply doing a git diff --exit-code and checking the exit code will do.
This action is more for pull requests where you don't know what is the base branch etc. and it is very very unstable.

File: .github/workflows/update_version_numbers.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        uses: tj-actions/changed-files@v34.6.1
+        with:
+          files: |
+             ./CMakeLists.txt
+             ./src/tests/class_tests/openms/source/VersionInfo_test.cpp
+             ./src/tests/topp/WRITE_INI_OUT.ini
+             ./src/tests/topp/*.ini
+
+# Commit updated files
+      - name: Commit updated files
```
Comment (jpfeuffer) [inline .github/workflows/update_version_numbers.yml:49]: Would be better if it does a pull request instead.

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
Thanks. Good structure!
MacOS should be fine. See comments.

File: .github/workflows/update_version_numbers.yml (hunk 1 above)
Comment (JeeH-K) [reply .github/workflows/update_version_numbers.yml:38]: That's a great idea! Thanks

**Comment by JeeH-K** [comment]
Seems like PR worked! https://github.com/JeeH-K/OpenMS/pull/8

**Comment by JeeH-K** [comment]
One more note: "on: push" is set for testing purposes. After approval, I'll remove it and set it to "workflow_dispatch"

**Comment by timosachsenberg** [comment]
I recall there was some ini file that one was not allowed to be updated because it was used in a test to do an ini file update to the newest version :)
We can try it after the merge of this PR and see if some tests fail I guess.

**Comment by jpfeuffer** [comment]
@JeeH-K Did you check that the removed Tex files are not used anywhere else?
Otherwise, revert the on: push to workflow dispatch, then it is good to go. Maybe we could also react to a push of a release/ tag @timosachsenberg

**Comment by JeeH-K** [comment]
> @JeeH-K Did you check that the removed Tex files are not used anywhere else? Otherwise, revert the on: push to workflow dispatch, then it is good to go. Maybe we could instead react to a push of a release/ tag @timosachsenberg

Tbh, no, I haven't checked. Since the latest changes were 2 years ago, I thought all the needed docs were already moved to the new repository... Frankly, I don't know how to check tho. Maybe I'll just revert them.

**Comment by timosachsenberg** [comment]
>Maybe we could instead react to a push of a release/ tag @timosachsenberg
hmm possible but would also be fine with a manual trigger. e.g, because maybe we then get a lot of PRs from different actions that in the end have conflicts?

**Comment by jpfeuffer** [comment]
@JeeH-K just git grep for the filenames. If they are referenced in code or other Tex/doxygen files.

@timosachsenberg how many actions create release tags? There should only be one and tags should be immutable.

**Comment by timosachsenberg** [comment]
I just mean if we also update the license text, changelog etc. with actions that trigger on the release tag and each of them opens a PR we might also need to manually merge one before the other.
But yeah let's try. More automatization the better :P

**Comment by jpfeuffer** [comment]
Why not put it all in here then?

**Comment by jpfeuffer** [comment]
Years should be updated every year. On develop.
Licenses should never be updated.
Change log should be done by GitHub, based on PRs with correctly formatted titles.

**Comment by JeeH-K** [comment]
> @JeeH-K just git grep for the filenames. If they are referenced in code or other Tex/doxygen files.

Thanks for the idea! All three files are not used in this repository.

**Comment by JeeH-K** [comment]
> Besides, years should be updated every year. On develop.

I believe another issue has been open for this: #6709

**Review by jpfeuffer — APPROVED** [review]

**Comment by jpfeuffer** [comment]
Ok let's merge. We can rename the action and add new triggers when we add more stuff.

**Comment by timosachsenberg** [comment]
thanks!

**Pull request merged**